GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -124,39 +124,112 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
+                // Set 'checkout.optimizeNewBranch=true' to enable optimized 'checkout -b'
                 { "checkout.optimizenewbranch", "true" },
+
+                // Needed to prevent line ending conversion. Prevents git from auto-replacing line endings with Crlf.
                 { "core.autocrlf", "false" },
+
+                // commit graph. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/299081
                 { "core.commitGraph", "true" },
+
+                // Perf - Git for Windows uses this to bulk-read and cache lstat data of entire directories (instead of doing lstat file by file).
                 { "core.fscache", "true" },
+
+                // Turns on all special gvfs logic. https://github.com/microsoft/scalar/blob/935c50110b31c440aae8e1cdc4ffea72bc3be01f/Scalar/CommandLine/ScalarVerb.cs#L35
                 { "core.gvfs", "true" },
+
+                // Use 'multi-pack-index' builtin instead of 'midx' to match upstream implementation
                 { "core.multiPackIndex", "true" },
+
+                // Perf - Enable parallel index preload for operations like git diff
                 { "core.preloadIndex", "true" },
                 { "core.safecrlf", "false" },
+
+                // Possibly cause hydration while creating untrackedCache.
                 { "core.untrackedCache", "false" },
                 { "core.repositoryformatversion", "0" },
+
+                // Support file modes onr Mac & Linux.
                 { "core.filemode", GVFSPlatform.Instance.FileSystem.SupportsFileMode ? "true" : "false" },
+
+                // For consistency with git init. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/235679
                 { "core.bare", "false" },
+
+                // For consistency with git init. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/235679
                 { "core.logallrefupdates", "true" },
+
+                // Git to download objects on demand.
                 { GitConfigSetting.CoreVirtualizeObjectsName, "true" },
+
+                // Configure hook that git calls to ask GVFS abvout what fiels it cares about
                 { GitConfigSetting.CoreVirtualFileSystemName, Paths.ConvertPathToGitFormat(GVFSConstants.DotGit.Hooks.VirtualFileSystemPath) },
+
+                // Ensure hooks path is configured correctly.
                 { "core.hookspath", expectedHooksPath },
+
+                // Hostname is no longer sufficent for VSTS authentication. VSTS now requires dev.azure.com/account to determine the tenant.
+                // By setting useHttpPath, credential managers will get the path which contains the account as the first parameter. They can then use this information for auth appropriately.
                 { GitConfigSetting.CredentialUseHttpPath, "true" },
                 { "credential.validate", "false" },
+
+                // This setting is not needed anymore, because current version of gvfs does not use index.lock.

```

Comment:
I would explain that we need to keep this here because the repo could have been cloned when this was set to `false` so we need to explicitly set it to `true` even though it is the default value.